tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -465,6 +465,60 @@ TEST(LinalgBackendEigen, SGMatrix_cross_entropy)
 	EXPECT_NEAR(ref, result, 1e-15);
 }
 
+TEST(LDA, SGMatrix_pinvh)
+{
+	SGMatrix<float64_t>A(3, 3);
+	A(0, 0) = 2.0;
+	A(0, 1) = -1.0;
+	A(0, 2) = 0.0;
+	A(1, 0) = -1.0;
+	A(1, 1) = 2.0;
+	A(1, 2) = -1.0;
+	A(2, 0) = 0.0;
+	A(2, 1) = -1.0;
+	A(2, 2) = 2.0;
+	SGMatrix<float64_t> I(3, 3);
+	linalg::identity(I);
+	SGMatrix<float64_t>B(3, 3);
+	linalg::pinvh(A, B);
+	SGMatrix<float64_t>C = linalg::matrix_prod(A, B);
+	C.display_matrix(C);
+	for(auto i:range(3))
+	{
+		for(auto j:range(3))
+		{
+			EXPECT_NEAR(I(i, j), C(i, j), 1e-14);
+		}
+	}
+}
+
+
+TEST(LDA, SGMatrix_pinv)
+{
+	SGMatrix<float64_t>A(2, 4);
+	A(0, 0) = 1.0;
+	A(0, 1) = 1.0;
+	A(0, 2) = 1.0;
+	A(0, 3) = 1.0;
+	A(1, 0) = 5.0;
+	A(1, 1) = 7.0;
+	A(1, 2) = 7.0;
+	A(1, 3) = 9.0;
+	SGMatrix<float64_t> I(2, 2);
+	linalg::identity(I);
+	SGMatrix<float64_t>B(4, 2);
+	linalg::pinvh(A, B);
```

Comment:
same comments for the test above regarding names, and explicitly checking the output